src/QsCompiler/Tests.Compiler/TestCases/TypeChecking.qs

```diff
@@ -579,5 +579,30 @@ namespace Microsoft.Quantum.Testing.TypeChecking {
         let fct = PartialApplication29();
         fct(1);
     }
-}
 
+    // Sized array constructors
+
+    function SizedArray1(n : Int) : Int[] {
+        return [10, size = n];
+    }
+
+    function SizedArray2() : String[] {
+        return ["foo", size = -1];
```

Comment:
What is the expectation with a negative integer size in the array constructor?

Reply:
Runtime exception. There are tests for that in microsoft/qsharp-runtime#507.